app/src/main/java/com/example/lhi06/residue/FindAlbumActivity.java

```diff
@@ -39,27 +41,31 @@ protected void onCreate(@Nullable final Bundle savedInstanceState) {
         recyclerView.setAdapter(adapter);
         service = new ItunesService();
 
+        adapter.setAlbumClickListener(new AlbumAdapter.AlbumClickListener() {
+            @Override
+            public void onClick(@NonNull final Album album) {
+                final Intent intent = new Intent(FindAlbumActivity.this, AlbumReviewActivity.class);
+                intent.putExtra(ALBUM_EXTRA, album);
+                startActivity(intent);
+            }
+        });
+
     }
 
     @OnClick(R.id.search_button)
     public void onSearchClicked() {
-        try {
-            service.getArtistsAlbums(editText.getText().toString(), new Callback<AlbumSearchResponse>() {
-                @Override
-                public void onResponse(final Call<AlbumSearchResponse> call, final Response<AlbumSearchResponse> response) {
-                    // on background thread
-                    adapter.setListContent(response.body().getResults());
-                }
-
-                @Override
-                public void onFailure(final Call<AlbumSearchResponse> call, final Throwable t) {
-                    final Context context = getApplicationContext();
-                    Toast.makeText(context, R.string.io_exception, Toast.LENGTH_SHORT).show();
-                }
-            });
+        service.getArtistsAlbums(artistNameEditText.getText().toString(), new Callback<AlbumSearchResponse>() {
+            @Override
+            public void onResponse(final Call<AlbumSearchResponse> call, final Response<AlbumSearchResponse> response) {
+                // on background thread
+                adapter.setListContent(response.body().getResults());
+            }
 
-        } catch (IOException e) {
-            Toast.makeText(this, R.string.io_exception, Toast.LENGTH_SHORT).show();
-        }
+            @Override
+            public void onFailure(final Call<AlbumSearchResponse> call, final Throwable t) {
+                final Context context = getApplicationContext();
```

Comment:
im not sure i think a while ago it was quite different and i had to add final to context and just havent changed it since